pkg/controllers/routing/network_routes_controller.go

```diff
@@ -1130,9 +1131,38 @@ func (nrc *NetworkRoutingController) startBgpServer(grpcServer bool) error {
 			}
 		}
 
+		// Get Global Peer Router LocalIP configs
+		var peerLocalIPs []string
+		nodeBGPPeerLocalIPs, ok := node.ObjectMeta.Annotations[peerLocalIPAnnotation]
+		if !ok {
+			klog.Infof("Could not find BGP peer local ip info in the node's annotations. Assuming node IP.")
+		} else {
+			peerLocalIPs = stringToSlice(nodeBGPPeerLocalIPs, ",")
+			err = func() error {
+				for _, s := range ipStrings {
```

Comment:
Did you mean to pass this as a parameter to the anonymous function?

Reply:
The anonymous function is there just to check if the list element is empty string or can be parsed as an IP address. The wrong list was being processed there, fixed now.